discord-scripts/thread-management/auto-join.ts

```diff
@@ -41,18 +43,27 @@ async function autoJoinThread(
   }
 
   const { guild: server, parent: containingChannel } = thread
-
   const placeholder = await thread.send("<placeholder>")
-
   // Use this to assign a specific role based on the mapping in CUSTOM_CHANNEL_ROLE, in order to map specific roles/channels
-  if (hasCustomChannels && containingChannel) {
-    const roleName = CUSTOM_CHANNEL_ROLE[containingChannel.name]
-    if (roleName) {
-      const channelMatchingRole = server.roles.cache.find(
-        (role) => role.name.toLowerCase() === roleName.toLowerCase(),
-      )
-      if (channelMatchingRole) {
-        await placeholder.edit(channelMatchingRole.toString())
+  if (containingChannel) {
+    const channelMapping = CUSTOM_CHANNEL_ROLE.find(
+      (mapping) => mapping.channelName === containingChannel.name,
+    )
+
+    if (channelMapping && channelMapping.roles.length > 0) {
+      const roleNames = channelMapping.roles
+
+      const rolesToTag = roleNames
+        .map((roleName) =>
+          server.roles.cache.find(
+            (role) => role.name.toLowerCase() === roleName.toLowerCase(),
+          ),
+        )
+        .filter((role): role is Role => role !== undefined)
```

Comment:
Slightly cleaner version we didn't use before:

```suggestion
      // make sure roleNames is all lowercased beforehand.

      const rolesToTag = roleNames
        .filter((role): role is Role =>
          roleNames.includes(role.name.toLowerCase()))
```